examples/c/profile.bpf.c

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+/* Copyright (c) 2022 Meta Platforms, Inc. */
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_core_read.h>
+
+#include "profile.h"
+
+char LICENSE[] SEC("license") = "Dual BSD/GPL";
+
+struct {
+	__uint(type, BPF_MAP_TYPE_RINGBUF);
+	__uint(max_entries, sizeof(struct stacktrace_event) * 64);
+} events SEC(".maps");
+
+SEC("perf_event")
+int profile(void *ctx)
+{
+	int pid = bpf_get_current_pid_tgid() >> 32;
+	int cpu_id = bpf_get_smp_processor_id();
+	struct stacktrace_event *event =
+		bpf_ringbuf_reserve(&events, sizeof(struct stacktrace_event), 0);
+
+	if (!event)
+		return 1;
+	event->pid = pid;
+	event->cpu_id = cpu_id;
+
+	event->kernel_stack_sz = bpf_get_stack(ctx, event->kernel_stack,
+					       MAX_STACK_DEPTH * sizeof(event->kernel_stack[0]),
```

Comment:
few problems here:

1. you are losing error code if error happens if you divide by sizeof(), do that division in user-space instead
2. use `sizeof(event->kernel_stack)` and don't duplicate MAX_STACK_DEPTH here. Again, if you change something about kernel_stack array you won't have to change this part.
3. Also just making sure (in case it's not Github UI formatting issue) that when wrapping function call, first wrapped argument on each line aligns with first argument on previous line.